homeassistant/components/wemo/light.py

```diff
@@ -308,26 +333,35 @@ def _update(self, force_update=True):
             if not self._available:
                 _LOGGER.info("Reconnected to %s", self.name)
                 self._available = True
-        except AttributeError as err:
+        except (AttributeError, ActionException) as err:
             _LOGGER.warning("Could not update status for %s (%s)", self.name, err)
             self._available = False
+            self.wemo.reconnect_with_device()
 
     def turn_on(self, **kwargs):
         """Turn the dimmer on."""
-        self.wemo.on()
-
-        # Wemo dimmer switches use a range of [0, 100] to control
-        # brightness. Level 255 might mean to set it to previous value
-        if ATTR_BRIGHTNESS in kwargs:
-            brightness = kwargs[ATTR_BRIGHTNESS]
-            brightness = int((brightness / 255) * 100)
-        else:
-            brightness = 255
-        self.wemo.set_brightness(brightness)
+        try:
+            self.wemo.on()
```

Comment:
Please move the turn on call down just above the set brightness call and just wrap those two calls with the try except block. We want to wrap as little as possible with the try except.